### PR TITLE
fix: twelve hygiene findings — canonical repo ids, honest figures, an audit surface that shows denials

### LINKED ISSUE #44: [P3] Round-2 hygiene: doc/code divergences, stale facts, dead surfaces, engine module split
## Findings

Low-severity findings from the round-2 review: doc/code divergences that fail safe, dead surfaces, stale facts, and one maintainability item. None is load-bearing; all are logged so they aren't lost. Several are one-line fixes.

### Doc/code divergences (all fail safe)

1. **Wave-0 auto-freeze is vaporware.** `docs/04-stations.md` and `PRODUCT.md` describe "at zero, Wave 0 may auto-freeze if policy is `owner` and lighting stays `lit`." `humanApprovalsRemaining` is decremented and displayed but **never read as a condition**; no `autoFreeze` code exists. Effect is *stricter* than documented (every packet keeps requiring a human), so it fails safe — but the doc describes a mechanism that isn't there. Either implement it or delete the claim. Related: status `frozen` is accepted by `approve` but set by no code path.

2. **ADR 0002 is stale.** It claims "scorecard treats abandoned drafts as closed-unmerged after 14 quiet days." The shipped, heavily-tested behavior deliberately does *not* write `closedUnmerged` at the quiet release — only a real GitHub close does (there's an explicit test asserting `closedUnmerged === 0` even at 45 quiet days). Correct the ADR so a future auditor's model of `health()` is right.

3. **"Stale-closed" reads as a fourth terminal bucket.** `docs/08-operations.md`'s merge-rate denominator lists "merged, closed-unmerged, or stale-closed" — in code there is no such bucket; a stale draft only becomes `closedUnmerged` when a human closes it. The arithmetic is unaffected; the wording implies a field that doesn't exist.

4. **README's absolute "never clicks merge"** sits alongside the self-disclosed `frontguard#196` operator merge. Disclosed as a doctrine miss everywhere else (PRODUCT §8, 06-v2, 08-operations, 03-allowlist, seed) — this is the honesty posture working — but the README line reads as a flat historical claim. Soften to match the rest.

5. **The negative-control refusal names a verb that doesn't exist.** It says "park the packet"; the operator's actual tool is `reject` (which sets status `rejected`, not `parked`). Fix the message.

### Stale facts

6. **"141 of 272 external PRs merged"** (background-agents, in `policy-records.json` + three docs, dated 2026-08-28) does not reproduce — a live query returns roughly 407 non-owner PRs / 250 merged. Re-derive with a stated methodology or drop the figure. Same discipline #3 applied to six other claims.

7. **From the market re-scan:** the "~40% of projects ban AI" figure carried in the field report has no traceable source; the real numbers are sharper (only ~11.8% of popular repos have *any* AI policy; ~3.6% hard-ban). The pydantic characterization ("bans mass submission + unassigned PRs") could not be confirmed from pydantic's own docs. Correct or soften both wherever they appear in project docs.

### Dead / inconsistent surfaces

8. **Dead code:** `applyQueueLive` and `scoutGithub` are fully unit-tested but never called from `cli.ts` (`tick` uses `tickWithGithub`/`applyTick` exclusively). Not a bug — but `applyQueueLive`'s `competingPr: boolean` collapses the two-tier competing/adjacent verdict into one flag, so wiring it up carelessly later would silently lose the human-triage distinction. Either delete or align it with `classifyCompetition`.

9. **`ledger` omits off-allowlist packets.** `status` says `packets=6`, `ledger` lists 5 — the denied `matplotlib/matplotlib` packet vanishes because the wave lookup goes through `repoById`. The audit surface under-reports exactly the denied work it should showcase.

10. **`isDenied` is case-insensitive; `repoById` is case-sensitive.** Verified *not* exploitable (a mismatch fails closed as "not on the allowlist"), but worth normalizing rather than relying on live paths always echoing the allowlist's own casing.

11. **`status` quiet-day counter drifts against the static seed.** `quiet=0d/14` is computed from the seed's frozen `updatedAt`, so on any day after the seed date it's wrong until a real `sync`. The seed is honestly a fixture; the operator-facing number should either be recomputed or labelled as seed-derived.

### Maintainability

12. **`engine.ts` (836 lines) and `engine.test.ts` (1303 lines) are monoliths** after sixteen units of accretion — lifecycle, evidence, competing-work, quiet-day, ledger-divergence and PR-body concerns in one file. Extracting `follow-up` (`applyPrSync` + quiet constants), `competing-work` (`classify*`), and the evidence gate into their own modules would make the seams explicit — and the seams are exactly where round 2 found the P0s.

## Acceptance

Each item either fixed, or explicitly closed with a recorded rationale. Items 1–5 and 9–11 are one-line-to-small; 6–7 need a re-derivation or a deletion; 8 and 12 are refactors that can wait for the next substantive change in those files.

### PR DESCRIPTION
**Pull request by ravidsrk**
Closes #44.

Twelve low-severity findings. The interesting part is not the fixes — it is what two of them broke on the way, and what that says about the rest.

### Two Criticals, both introduced by the first attempt at this issue
- **Item 10's case-normalisation turned a loud fail-closed into a silent fail-open.** `repoById` was normalised; three scorecard lookups were not. The operator's same-hour stop — `halt <repoId>` when a maintainer asks — began reporting success, incrementing `bans`, and halting nothing. `main` had refused loudly; the fix made it lie quietly. Now canonicalised once at the boundary, so `halt` in any casing resolves to the roster's spelling and genuinely moves the row, parks the packet, and counts the ban. An unknown or denylisted id is still refused loudly at exit 1.
- **Item 6's re-derivation was written into `policy-records.json`'s `quote`.** That field renders on the maintainer-facing evidence page under *"What your policy says / From `CONTRIBUTING.md`"* — as a blockquote attributed to **their** file. A background-agents maintainer would have been shown Foundry's GitHub search query and our internal issue number as their own words. `docs/SPEC.md` §3 reserves that field for one verbatim statement. The quote is back to the absence note alone; the figure and its method live in `allowlist.yaml`'s `policyNotes`, where the schema puts provenance, and `parsePolicyRecords` now **refuses** a `silent` record whose quote carries a derived figure — so the convention is enforced rather than stated.

### The figures
`141 of 272` did not reproduce. Re-derived live and recorded with its method and date: **250 of 408 non-owner PRs merged (61%)**, `repo:ColeMurray/background-agents is:pr -author:ColeMurray`, measured 2026-08-29. Confirmed independently three times during review.

Two of the issue's own claims were **refuted and deliberately not acted on**: the pydantic characterisation checks out against the live source, and the "~40% of projects ban AI" figure appears nowhere in this repo. The issue text was also wrong in two places — `stale-closed` was in three files, not one, and `scoutGithub` is not "fully unit-tested" but has zero references.

### The audit surface
`status` said `packets=6` while `ledger` listed 5 — the denied `matplotlib/matplotlib` packet vanished through a wave lookup. The surface under-reported exactly the denied work it exists to showcase. Now grouped under "Off allowlist — denied or unlisted", with the counts asserted to agree and the committed GENERATED block byte-identical to the command.

### On the provenance claim
A round-2 finding called a pydantic file-layout description a false citation. It was not. The root `CONTRIBUTING.md` is a **git symlink** — tree mode `120000`, a 20-byte blob reading `docs/contributing.md`, which the Contents API resolves to 12,547 bytes. Both earlier descriptions were true of the same object through different surfaces. The wording now names the mechanism, so neither can recur.

### Verification at `9cbd58d`
- **165/165**, twice from a clean tree. `npm run validate` exit 0.
- Three review rounds, three isolated axes. Round 3: standards+spec PASS; test-adequacy FAIL on one coverage item, tracked as #57.
- All eight canonicalisation pieces die separately under mutation; both over-fit probes caught; the quote guard is mutation-proven in both directions and does not fire on a reworded absence note.

### Known residual, disclosed
`factory/engine.ts:285` and `:295` survive reverting to raw `===` with the suite green. The shipped code is correct — this is a coverage hole, tracked as **#57** with the exact fixture the fix needs. Under this sweep's round budget the unit should have been parked; the operator was asked and chose to merge the verified work and track it. Recorded as a deliberate exception.

Item 12 (the `engine.ts` module split) is declined with a recorded rationale: two other units were editing that file concurrently, and a large no-behaviour-change move is exactly the diff that hides a regression.

<details open><summary><h3>Greptile Summary</h3></summary>

The PR corrects repository-ID handling, policy provenance, status-ledger visibility, scorecard terminology, and related documentation. The main implementation changes are:
- Canonicalize repository identities across packet, halt, allowlist, and scorecard paths.
- Include denied and unlisted packets in generated ledger sections and reconcile displayed counts.
- Reject derived figures in silent policy-record quotes, although the new lexical guard has two non-blocking edge cases.
- Expand regression coverage and align operational documentation with implemented behavior.
</details>

<details open><summary><h3>Confidence Score: 4/5</h3></summary>

The PR appears safe to merge, with two non-blocking edge cases in the new silent-policy quote heuristic.

Canonical repository handling and denied-packet ledger visibility are consistent across their callers, while the new provenance regex remains incomplete for capitalization variants and overly broad for numeric path components.

**Files Needing Attention:** factory/policy-records.ts, factory/policy.test.ts
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| factory/allowlist.ts | Adds ASCII case-insensitive repository comparison and canonical roster-ID resolution, with duplicate and denylist protections. |
| factory/engine.ts | Applies canonical repository identities to halt, packet, and scorecard flows; investigated lookup paths remain consistent. |
| factory/status.ts | Groups every packet into a wave or off-allowlist section so denied work remains visible and countable. |
| factory/policy-records.ts | Adds a silent-quote provenance guard, but its regex misses capitalization variants and can reject numeric path segments. |
| factory/policy.test.ts | Covers common derived-figure and legitimate absence-note forms but omits capitalization and numeric-path edge cases. |
| factory/engine.test.ts | Adds broad regression coverage for canonical IDs, ledger count agreement, generated output, and observation dates. |
| policy-records.json | Restores the silent quote to an absence statement and keeps behavioral measurements out of attributed source text. |

</details>

<details open><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
  A[Repository or CLI identifier] --> B[Canonical allowlist lookup]
  B --> C[Packet and scorecard operations]
  C --> D[Factory state]
  D --> E[Status and ledger sections]
  P[Policy records] --> Q[Parse and provenance guard]
  Q --> R[Policy evaluation]
```
</details>

<a href="https://app.greptile.com/ide/claude-code?prompt=Greploop%20ravidsrk%2Foss-foundry%20PR%20%2358%3A%20work%20through%20Greptile's%20open%20review%20comments%2C%20then%20keep%20reviewing%20and%20fixing%20until%20it%20comes%20back%20clean%20at%205%2F5%20with%20zero%20unresolved%20comments.%0AStart%20by%20reading%20the%20comments%20off%20the%20PR%20itself.%20On%20GitHub%2C%20use%20paginated%20%60gh%20api%20graphql%60%20to%20query%20%60pullRequest.reviewThreads%60%20with%20each%20thread's%20%60isResolved%60%20value%20and%20inline%20%60comments%60.%20%60gh%20pr%20view%20--comments%60%20only%20includes%20conversation%20comments%2C%20so%20do%20not%20use%20it%20as%20the%20findings%20source.%20They%20are%20not%20listed%20in%20this%20prompt%20on%20purpose%3A%20the%20PR%20is%20current%2C%20a%20pasted%20copy%20would%20not%20be.%20Skip%20anything%20already%20resolved%2C%20and%20if%20you%20judge%20a%20comment%20wrong%2C%20say%20so%20rather%20than%20changing%20code%20to%20satisfy%20it.%0A%0APrefer%20the%20Greptile%20CLI%2C%20which%20reviews%20the%20working%20tree%20with%20no%20push%20and%20no%20CI%20run.%20Fall%20back%20to%20PUSH%20LOOP%20only%20where%20a%20step%20below%20says%20to.%0A1.%20Run%20%60command%20-v%20greptile%60.%20Missing%3A%20go%20to%20PUSH%20LOOP%2C%20and%20pitch%20the%20CLI%20at%20the%20end.%0A2.%20Run%20%60greptile%20whoami%60.%20It%20exits%200%20even%20when%20signed%20out%2C%20so%20read%20the%20output%3A%20if%20it%20says%20%22Not%20signed%20in%22%2C%20ask%20the%20user%20to%20run%20%60greptile%20login%60%20%28it%20opens%20a%20browser%2C%20do%20not%20run%20it%20yourself%29%20and%20go%20to%20PUSH%20LOOP%20with%20no%20pitch.%0A3.%20Look%20for%20%60greploop%2FSKILL.md%60%20under%20%60.agents%2Fskills%2F%60%20or%20%60.claude%2Fskills%2F%60%2C%20both%20in%20the%20repo%20root%20and%20in%20the%20user's%20home%20directory.%20Found%3A%20follow%20that%20skill%2C%20never%20pushing%20between%20rounds%2C%20and%20you%20are%20done.%0A4.%20Missing%3A%20run%20%60greptile%20skills%20list%60.%20If%20it%20fails%20as%20an%20unknown%20command%2C%20this%20CLI%20predates%20the%20skill%2C%20so%20tell%20the%20user%20to%20run%20%60greptile%20update%60%20and%20go%20to%20PUSH%20LOOP%20with%20no%20pitch.%20Otherwise%20ask%20the%20user%20before%20running%20%60greptile%20skills%20install%20greploop%60%2C%20then%20follow%20the%20skill%20it%20writes.%0A%0APUSH%20LOOP%20%28only%20when%20a%20step%20above%20sends%20you%20here%29%3A%0A%0AGREPLOOP%20ENABLED%3A%20After%20fixing%20all%20issues%2C%20commit%20and%20push%20your%20changes.%20Greptile%20will%20automatically%20re-review%20the%20PR%20via%20webhook.%20Wait%20for%20the%20review%20to%20complete%2C%20then%20read%20the%20new%20review%20comments%20and%20confidence%20score.%20If%20the%20confidence%20score%20is%20below%205%2F5%20or%20there%20are%20unresolved%20comments%2C%20fix%20those%20issues%20and%20push%20again.%20Repeat%20until%20the%20confidence%20score%20is%205%2F5%20with%20zero%20unresolved%20comments.%20Maximum%205%20iterations.%20Do%20NOT%20open%20a%20new%20PR%20%E2%80%94%20keep%20pushing%20to%20the%20same%20branch.%0A%0AIF%20YOU%20ENDED%20UP%20ON%20PUSH%20LOOP%20BECAUSE%20THE%20CLI%20WAS%20MISSING%3A%20once%20the%20loop%20is%20done%2C%20print%20the%20block%20below%20and%20nothing%20else%20about%20the%20CLI.%20Reproduce%20it%20character%20for%20character%20inside%20a%20fenced%20code%20block%2C%20so%20the%20borders%20line%20up.%20Do%20not%20reword%20it%2C%20translate%20it%2C%20summarize%20it%2C%20add%20to%20it%2C%20or%20drop%20the%20box.%20Print%20it%20once%2C%20and%20do%20not%20bring%20the%20CLI%20up%20again%20later%20in%20the%20session.%0A%0A%2B----------------------------------------------------------------------------%2B%0A%7C%20%20You%20can%20run%20greploops%20faster%20locally%20with%20our%20CLI.%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7C%0A%7C%20%20Install%20it%20at%20https%3A%2F%2Fwww.greptile.com%2Fcli%2C%20or%20I%20can%20install%20it%20for%20you.%20%20%7C%0A%2B----------------------------------------------------------------------------%2B%0A%0AIf%20they%20take%20you%20up%20on%20it%2C%20install%20with%20%60npm%20install%20-g%20greptile%60%20%28or%20%60brew%20install%20greptileai%2Ftap%2Fgreptile%60%29%2C%20then%20%60greptile%20skills%20install%20greploop%60.%20Leave%20%60greptile%20login%60%20to%20them%2C%20it%20opens%20a%20browser.&repo=ravidsrk%2Foss-foundry&pr=58&platform=github">Fix all with Greploop</a> <a href="https://app.greptile.com/ide/claude-code?prompt=%23%23%23%20Issue%201%0Afactory%2Fpolicy-records.ts%3A13%0A**Capitalized%20ratios%20bypass%20guard**%0A%0AThe%20ratio%20alternative%20matches%20only%20lowercase%20%60of%60%2C%20so%20a%20silent%20quote%20such%20as%20%60141%20Of%20272%20external%20PRs%20merged%60%20passes%20the%20new%20guard%20and%20can%20still%20render%20a%20derived%20figure%20as%20the%20maintainer's%20own%20words.%0A%0A%23%23%23%20Issue%202%0Afactory%2Fpolicy-records.ts%3A13%0A**Numeric%20paths%20trigger%20ratio%20guard**%0A%0AThe%20unrestricted%20%60number%2Fnumber%60%20alternative%20also%20matches%20legitimate%20path%20segments%20such%20as%20%60docs%2F2026%2F08%60%2C%20causing%20policy-record%20validation%20and%20lazy%20policy%20loading%20to%20reject%20an%20otherwise%20valid%20silent%20absence%20note.%0A%0A---%0A%0AFor%20each%20issue%20above%2C%20determine%20whether%20it%20is%20valid%20and%20should%20be%20fixed.%20If%20so%2C%20fix%20it%20directly.&repo=ravidsrk%2Foss-foundry&pr=58&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaudeDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaude.svg?v=6">Fix All in Claude Code</picture></a>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
### Issue 1
factory/policy-records.ts:13
**Capitalized ratios bypass guard**

The ratio alternative matches only lowercase `of`, so a silent quote such as `141 Of 272 external PRs merged` passes the new guard and can still render a derived figure as the maintainer's own words.

### Issue 2
factory/policy-records.ts:13
**Numeric paths trigger ratio guard**

The unrestricted `number/number` alternative also matches legitimate path segments such as `docs/2026/08`, causing policy-record validation and lazy policy loading to reject an otherwise valid silent absence note.

---

For each issue above, determine whether it is valid and should be fixed. If so, fix it directly.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["fix: round-2 corrections — provenance, a..."](https://github.com/ravidsrk/oss-foundry/commit/9cbd58de97764b6fd7f45ef9ba9a7946fe48ff53) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=58168936)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: factory/policy-records.ts [hunk 1]
```diff
@@ -5,6 +5,13 @@ import type { PolicyRecord } from "./types.ts";
 
 const STANCES = new Set(["forbidden", "conditional", "welcome", "silent"]);
 
+/**
+ * A derived measurement: a ratio (`141 of 272`, `141/272`) or a percentage (`61%`). Deliberately
+ * narrow — an ISO date, a file path and a version number must all pass, because absence notes carry
+ * them ("no CONTRIBUTING.md ... as of 2026-08-28").
+ */
+const DERIVED_FIGURE = /\b\d[\d,]*\s*(?:\/|of)\s*\d[\d,]*\b|\b\d+(?:\.\d+)?\s*%/;
```
[the remark this answers is not in the document]
Comment (ravidsrk) [reply factory/policy-records.ts:13]: Already covered. The derived-figure guard is case-insensitive (`factory/policy.test.ts`: "the derived-figure guard reads case-insensitively and does not mistake a path for a ratio").

[the remark this answers is not in the document]
Comment (ravidsrk) [reply factory/policy-records.ts:13]: Already covered. Same test: a numeric path is not a ratio. The committed policy records parse under that guard.
